src/pages/Share/AddFile.tsx

```diff
@@ -44,51 +46,54 @@ const useStyles = makeStyles((theme: Theme) =>
   }),
 )
 
-export default function Upload({ setFile }: Props): ReactElement {
+export default function Upload({ setFiles }: Props): ReactElement {
   const classes = useStyles()
+  const ref = useRef<HTMLInputElement>(null)
   const history = useHistory()
   const [isDragging, setIsDragging] = useState<boolean>(false)
   const [agreedToTerms, setAgreedToTerms] = useState(Boolean(window.localStorage.getItem('agreedToTerms')))
 
+  useEffect(() => {
+    if (ref.current !== null) {
+      ref.current.setAttribute('directory', '')
+      ref.current.setAttribute('mozdirectory', '')
+      ref.current.setAttribute('webkitdirectory', '')
+    }
+  }, [ref])
+
   const handleAgree = () => {
     setAgreedToTerms(true)
     window.localStorage.setItem('agreedToTerms', Date.now().toString())
   }
 
   const onDragOver = (ev: DragEvent<HTMLDivElement>) => {
-    setIsDragging(true)
-
     ev.preventDefault()
+    ev.stopPropagation()
+
+    setIsDragging(true)
   }
 
-  const onDrop = (ev: DragEvent<HTMLDivElement>) => {
+  const onDrop = async (ev: DragEvent<HTMLDivElement>) => {
     // Prevent default behavior (Prevent file from being opened)
     ev.preventDefault()
+    ev.stopPropagation()
 
-    if (ev.dataTransfer.items) {
-      const fls = []
-      // Use DataTransferItemList interface to access the file(s)
-      for (let i = 0; i < ev.dataTransfer.items.length; i++) {
-        // If dropped items aren't files, reject them
-        if (ev.dataTransfer.items[i].kind === 'file') {
-          const fl = ev.dataTransfer.items[i].getAsFile()
-
-          if (fl) fls.push(fl)
-        }
-      }
-
-      if (fls.length === 1) setFile(fls[0])
-    } else {
-      // Use DataTransfer interface to access the file(s)
-      if (ev.dataTransfer.files.length === 1) setFile(ev.dataTransfer.files[0])
-    }
+    setFiles(await handleDrop(ev))
     setIsDragging(false)
   }
 
-  const onDragLeave = (ev: DragEvent<HTMLDivElement>) => {
-    setIsDragging(false)
+  const handleFiles = (files: FileList | null) => {
+    const fls = []
 
+    if (files) for (let i = 0; i < files.length; i++) fls.push(new SwarmFile(files[i]))
+    setFiles(fls)
+  }
```

Comment:
Ah I see why you have to iterate, but maybe calling the parameter `fileList` and the array `files` is a bit better so the naming is not confusing 🙂